core/common/src/main/java/alluxio/collections/LockPool.java

```diff
@@ -256,7 +269,9 @@ private Resource getResource(K key) {
         v.mIsAccessed = true;
         return v;
       }
-      return new Resource(mDefaultLoader.apply(k));
+      Resource res = mResueLockPool.acquire();
```

Comment:
the behavior will be changed here, right? if we run out of the `mReuseLockPool`, will this `getResource` block?

Reply:
Yeah, it could be blocked